packages/examples/packages/interactive-ui/src/index.ts

```diff
@@ -0,0 +1,201 @@
+import { rpcErrors } from '@metamask/rpc-errors';
+import type {
+  OnHomePageHandler,
+  OnTransactionHandler,
+  OnUserInputHandler,
+} from '@metamask/snaps-sdk';
+import {
+  panel,
+  type OnRpcRequestHandler,
+  heading,
+  button,
+  UserInputEventType,
+  form,
+  input,
+  ButtonType,
+  text,
+  copyable,
+} from '@metamask/snaps-sdk';
+
+/**
+ * Initiate a new interface with the starting screen and return the Snap interface ID.
+ *
+ * @returns The Snap interface ID.
+ */
+export const createInterface = async (): Promise<string> =>
+  snap.request({
+    method: 'snap_createInterface',
+    params: {
+      ui: panel([
+        heading('Interactive UI Example Snap'),
+        button({ value: 'Update UI', name: 'update' }),
+      ]),
+    },
+  });
+
+/**
+ * Update the interface with a simple form containing an input and a submit button.
+ *
+ * @param id - The Snap interface ID to update.
+ */
+export const showForm = async (id: string) => {
+  snap.request({
+    method: 'snap_updateInterface',
+    params: {
+      id,
+      ui: panel([
+        heading('Interactive UI Example Snap'),
+        form({
+          name: 'example-form',
+          children: [
+            input({ name: 'example-input', placeholder: 'Enter something...' }),
+            button('Submit', ButtonType.Submit, 'sumbit'),
+          ],
+        }),
+      ]),
+    },
+  });
+};
+
+/**
+ * Update a Snap interface to show a given value.
+ *
+ * @param id - The Snap interface ID to update.
+ * @param value - The value to display in the UI.
+ */
+export const showResult = async (id: string, value: string) => {
+  snap.request({
+    method: 'snap_updateInterface',
+    params: {
+      id,
+      ui: panel([
+        heading('Interactive UI Example Snap'),
+        text('The submitted value is:'),
+        copyable(value),
+      ]),
+    },
+  });
+};
+
+/**
+ * Handle incoming JSON-RPC requests from the dapp, sent through the
+ * `wallet_invokeSnap` method. This handler handles two methods:
+ *
+ * - `dialog`: Create a `snap_dialog` with an interactive interface. This demonstrates
+ * that a snap can show an interactive `snap_dialog` that the user can interact with.
+ *
+ * - `get_state`: Get the state of a given interface. This demonstrates
+ * that a snap can retrieve an interface state.
+ *
+ * @param params - The request parameters.
+ * @param params.request - The JSON-RPC request object.
+ * @returns The JSON-RPC response.
+ * @see https://docs.metamask.io/snaps/reference/exports/#onrpcrequest
+ * @see https://docs.metamask.io/snaps/reference/rpc-api/#wallet_invokesnap
+ */
+export const onRpcRequest: OnRpcRequestHandler<{ id: string }> = async ({
+  request,
+}) => {
+  switch (request.method) {
+    case 'dialog': {
+      const interfaceId = await createInterface();
+
+      snap.request({
```

Comment:
Will the snap not terminate by being idle if you do this instead of awaiting?

Reply:
I don't know but I can't wait for the confirmation to resolve to return the interface ID, I need it while the confirmation is openend. I could store the interface ID and have another method to get it.

Reply:
Yeah, right now this snap would be consider idle because we only consider inbound requests when determining if the snap is idle or not.